pyproject.toml

```
@@ -84,23 +84,47 @@ line-length = 99
 
 [tool.ruff.lint]
 explicit-preview-rules = true
-select = ["A", "E", "W", "F", "C", "N", "D", "I", "CPY001"]
+select = [
+  "A",
+  "E",
+  "W",
+  "F",
+  "C",
+  "N",
+  "D",
+  "I",
+  "B",
+  "CPY001",
+  "RUF",
+  "S",
+  "SIM",
+  "UP",
+  "TC",
+]
 ignore = [
-    # Missing docstring in public method (pydocstyle doesn't look for docstrings in super class
-    # https://github.com/PyCQA/pydocstyle/issues/309) TODO: add pylint check? https://github.com/PyCQA/pydocstyle/issues/309#issuecomment-1284142716
-    "D102",
-    "D105", # Missing docstring in magic method
-    "D107", # Missing docstring in __init__
-    "D403", # First word of the first line should be capitalized (false positive on "MySQL")
-    "D415", # Docstring first line punctuation (doesn't make sense for properties)
-    "E501", # Line too long (because using black creates errors with this)
-    "N818", # Exception name should be named with an Error suffix
-    "W505", # Doc line too long (so that strings in comments aren't split across lines)
+  # Missing docstring in public method (pydocstyle doesn't look for docstrings in super class
+  # https://github.com/PyCQA/pydocstyle/issues/309) TODO: add pylint check? https://github.com/PyCQA/pydocstyle/issues/309#issuecomment-1284142716
+  "D102",
+  "D105", # Missing docstring in magic method
+  "D107", # Missing docstring in __init__
+  "D403", # First word of the first line should be capitalized (false positive on "MySQL")
+  "D415", # Docstring first line punctuation (doesn't make sense for properties)
+  "E501", # Line too long (because using black creates errors with this)
+  "N818", # Exception name should be named with an Error suffix
+  "W505", # Doc line too long (so that strings in comments aren't split across lines)
+  "S101",
```

```suggestion
  "S101",
  "B011",
  "B903",
  "B904",  # Exception chaining happens automatically if `from` omitted
  "RUF005",
  "RUF010",
```
b011 for same reason as s101
b903 class with only `__init__` shouldn't always be a dataclass
b904 unnecessarily verbose, default behavior of exception chaining (when `from` omitted) is almost always desired
ruf005 imo premature optimization; sometimes less readable
ruf010 imo decreases readability

---

👋🏻 My 2 cents here highly varies from rule to rule:

- [`B011`](https://docs.astral.sh/ruff/rules/assert-false/) - Unnecessary if `S101` gets apply, which I am in favour of.
- [`B903`](https://docs.astral.sh/ruff/rules/class-as-data-structure/) - No preference.
- [`B904`](https://docs.astral.sh/ruff/rules/raise-without-from-inside-except/) - No preference, although if exception names are fine-grained enough, I will say unnecessay.
- [`RUF005`](https://docs.astral.sh/ruff/rules/collection-literal-concatenation/) - I like it, and find it more readable than concatenating using `+`.
- [`RUF010`](https://docs.astral.sh/ruff/rules/explicit-f-string-type-conversion/) - No preference.